tools/readme_linter/assert.go

```diff
@@ -0,0 +1,128 @@
+package main
+
+import (
+	"fmt"
+	"regexp"
+	"sort"
+
+	"github.com/yuin/goldmark/ast"
+)
+
+//type for all linter assert methods
+type T struct {
+	filename       string
+	markdown       []byte
+	newlineOffsets []int
+
+	fails int
+}
+
+// called by all assert functions that involve a node
+func (t *T) printFailedAssertf(n ast.Node, format string, args ...interface{}) {
+	t.printFile(n)
+	fmt.Printf(format+"\n", args...)
+	//t.printRule(3)
+	t.fails++
+}
+
+// Assert function that doesnt involve a node, for example if something is missing
+func (t *T) assertf(format string, args ...interface{}) {
+	t.assertLinef(0, format, args...) // There's no line number associated, so use the first
+}
+
+func (t *T) assertNodef(n ast.Node, format string, args ...interface{}) {
+	t.printFailedAssertf(n, format, args...)
+}
+
+func (t *T) assertLinef(line int, format string, args ...interface{}) {
+	t.printFileLine(line)
+	fmt.Printf(format+"\n", args...)
+	//t.printRule(2) This can either be 2 or 3 since it's called through assertf
+	t.fails++
+}
+
+// func (t *T) printRule(callers int) {
+// 	pc, codeFilename, codeLine, ok := runtime.Caller(callers)
+// 	if !ok {
+// 		panic("can't get caller")
+// 	}
+
+// 	f := runtime.FuncForPC(pc)
+// 	var funcName string
+// 	if f != nil {
+// 		funcName = f.Name()
+// 	}
+
+// 	fmt.Printf("%s:%d: ", codeFilename, codeLine)
+// 	if len(funcName) == 0 {
+// 		fmt.Printf("failed assert\n")
+// 	} else {
+// 		fmt.Printf("failed assert in function %s\n", funcName)
+// 	}
+// }
+
+func (t *T) line(offset int) int {
+	return sort.SearchInts(t.newlineOffsets, offset)
+}
+
+func (t *T) printFile(n ast.Node) {
+	lines := n.Lines()
+	if lines.Len() == 0 {
+		panic("can't get offset of node")
```

Comment:
I hit this panic when running `./readme_linter ../../plugins/inputs/bind/README.md `, looks like it hit when trying to assert for a fenced code block `t.assertNodef(toml, "config next sibling isn't a fenced code block")`. Maybe this is similar to the fenced code block issue I faced with the sample config tool?

Reply:
I changed the code to use the first line in the file in this case. That's not ideal, but at least it doesn't panic. I don't know if there's a way in goldmark to find the original line of a node like a fenced code block.